### PR TITLE
Ralph Mallett - bowling challenge

### PR DESCRIPTION
**Pull request by ralphm10**
Partial Solution

- complete for one frame only 
-  includes tests for various edge cases
- need to create separate functions (or class) for frame 10
- next step would be to have frame and game classes interact, and create functions to add spare/strike bonus

### CODE REVIEW

File: spec/FrameSpec.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    });
+
+  describe ('Scoring', function () {
+    it ('returns 0 for two gutter balls', function() {
+      frame.enterPins(0);
+      frame.enterPins(0);
+      expect(frame.getScore()).toEqual(0);
+    });
+
+    it ('returns score for two scoring rolls', function() {
```
Comment (02ship) [inline spec/FrameSpec.js:18]: Clear progression from the simplest possible version, very readable, gives an overview of your development process

File: spec/FrameSpec.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      frame.enterPins(1);
+      frame.enterPins(2);
+      expect(frame.getScore()).toEqual(3);
+    });
+
+    it ('scores 0 if a strike is thrown', function() {
```
Comment (02ship) [inline spec/FrameSpec.js:24]: I hadn't thought about using proper bowling notation for strikes and spares, what advantages/disadvantages does this offer?

File: spec/GameSpec.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        }
+        expect(game.getScore()).toEqual(0);
+      });
+
+
+      it ('returns 20 for a game of 1s', function() {
+        for (var i = 0; i < 10; i++) {
+          game.frameScores.push(2);
+        }
+        expect(game.getScore()).toEqual(20);
```
Comment (02ship) [inline spec/GameSpec.js:25]: Very logical test progression, makes it obvious what the code is currently capable of handling

File: spec/FrameSpec.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    it ('cannot enter a strike as second roll', function() {
+      frame.enterPins(5);
+      expect(frame.enterPins('X')).toEqual('Second roll cannot be a strike!')
+    });
+
+    it ('cannot enter more than 2 rolls in a standard frame', function() {
+      frame.enterPins(1);
+      frame.enterPins(2);
+      expect(frame.enterPins(2)).toEqual('Frame closed!')
+    });
```
Comment (02ship) [inline spec/FrameSpec.js:85]: Love the edge case tests

File: spec/SpecHelper.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+beforeEach(function () {
+  jasmine.addMatchers({
```
Comment (02ship) [inline spec/SpecHelper.js:2]: Left in as a template for your own custom matcher?

File: src/Frame.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+class Frame {
+  constructor(){
+    this.frameNumber = 1;
+    this.frameScore = 0;
+    this.frameRolls = [];
+    this.MAX_NORMAL_FRAME = 2;
+    this.PINS = 10;
+    this.VALID_INPUTS = [0,1,2,3,4,5,6,7,8,9,'X','/']
```
Comment (02ship) [inline src/Frame.js:8]: Great use of constants to constrain your code

File: src/Game.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  addFrame(frame) {
+    if (frame.isOpen() === false) { game.push(frame.getScore())
+    }
+  }
+
+  // need to implement this - perhaps it moves to frame class
+
+  getScore() {
```
Comment (02ship) [inline src/Game.js:13]: I'm not sure I follow what 'a' and 'b' are here, are they the first and second rolls of a frame?

File: src/Frame.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
+class Frame {
+  constructor(){
```
Comment (DillonBarker) [inline src/Frame.js:2]: Can you remove any more magic numbers?
